docs/changelog/0.4.0.md

```diff
@@ -0,0 +1,528 @@
+
+Using methods that directly or transitively need the removed classes will require an additional dependency on the appropriate third-party library.
+For example:
+```scala
+val str: String = "Hello Native"
+str.toLowerCase()              // works as before
+str.toLowerCase(Locale.French) // requires scala-java-locales to link
+```
+
+### `NativeConfig` replaces setting keys 
+The `nativeXyz` setting keys are now deprecated in favor of a single `nativeConfig` setting, which can be used as follows:
+```scala
+// build.sbt
+nativeConfig ~= {
+  _.withMode(build.Mode.releaseFast)
+   .withGC(build.GC.immix)
+   .withLTO(build.LTO.full)
+   .withOptimize(true)
+   .withCompileOptions(Nil)
+   .withLinkingOptions(Nil)
+}
+```
+Old style settings keys are still supported, but they have lower priority than the new config and will be removed 
+at some point in the future.
+In the following example resulting `LTO` setting would be set to `thin`
+```scala
+nativeConfig := nativeConfig.value.withLTO(build.LTO.thin)
+
+nativeLTO := "none"
+```
+
+### CFuncPtr changes
+You no longer need to implement the `CFuncPtrN` trait which is now private for Scala Native implementation.
+Instead, you can use an implicit conversion method taking arbitrary `scala.FunctionN` and returning `CFuncPtrN`.
+
+```scala
+type Callback = CFuncPtr1[CInt,Unit]
+def registerCallback(cFn: Callback): Unit = extern
+def fn(n: CInt): Unit = ???
+
+registerCallback(CFuncPtr1.fromScalaFunction(fn))
+registerCallback(fn)
+
+registerCallback { (n: CInt) => println("hello native") }
+```
+
+It's now also possible to work with an arbitrary pointer and convert it to `CFuncPtrN` that can be called in your Scala code
+or to convert your function to any pointer if your native library needs this.
+
+```scala
+import scala.scalanative.unsafe.Ptr
+val cFnPtr: CFuncPtr0[CInt]    = ???
+val fnPtr: Ptr[Byte]           = CFuncPtr.toPtr(cFnPtr)
+val fnFromPtr: CFuncPtr0[CInt] = CFuncPtr.fromPtr[CFuncPtr0[CInt]](fnPtr)
+```
+
+### Other breaking changes:
+* Sbt 0.13.x is no longer supported - upgrade to 1.1.6 or newer.
+* The minimal version of Clang working with Scala Native is now 6.0
+* `CSize` is now unsigned numeral type
+* Usage of signed numeral types for methods expecting `CSize` was deprecated. 
+
+## New features
+
+### Supported Scala versions
+We added support for Scala 2.12.13 and 2.13.4, in addition to the existing support for 2.11.12. 
+
+### JUnit Support
+Scala Native now comes with JUnit support out of the box, this means that you can write tests in the same way you would do
+for a Scala/JVM or Scala.js project. To enable JUnit tests all you will need to do is to add the two following lines to your `build.sbt`.
+
+```scala
+addCompilerPlugin("org.scala-native" % "junit-plugin" % SNVersion cross CrossVersion.full)
+libraryDependencies += "org.scala-native" % "junit-runtime" %%% SNVersion % "test"
```

Comment:
Have you considered `nativeVersion.value` here?
```suggestion
addCompilerPlugin("org.scala-native" % "junit-plugin" % nativeVersion.value cross CrossVersion.full)
libraryDependencies += "org.scala-native" % "junit-runtime" %%% nativeVersion.value % "test"
```

Reply:
How do you get `nativeVersion.value`?

Reply:
@ekrich `nativeVersion` is [generated by the Scala Native sbt plugin](https://github.com/scala-native/scala-native/blob/783dda1b38abea77509e52b68b0f1b55089b288a/sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePlugin.scala#L15).
I wrote `nativeVersion.value` first because I didn't remember correctly.